qiita_db/handlers/sample_information.py

```diff
@@ -0,0 +1,65 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from tornado.web import HTTPError
+
+import qiita_db as qdb
+from .oauth2 import OauthBaseHandler, authenticate_oauth
+
+
+def _get_sample_info(sid):
```

Comment:
Looks like `_get_user_info` and `_get_sample_info` are essentially the same function except for the class instantiating it. I would probably try to facto these two things into a single function and then make `_get_sample_info` a shortcut for a call to this new function. Something like:

```python
def _get_instance(object_id, klass, reason):
    """ """
    try:
        object = klass(object_id)
    except qdb.exceptions.QiitaDBUnknownIDError:
        raise HTTPError(404)
    except Exception as e:
        raise HTTPError(500, reason=reason + ' %s: %s' % (object_id, str(e)))


def _get_study_info(sid):
    return _get_instance(email, qdb.user.User, 'Error instantiating user')

def _get_study_info(sid):
    return _get_instance(int(sid), qdb.metadata_template.sample_template.SampleTemplate, 'Error instantiating sample information')

```

Should ultimately reduce needed code and tests 👍